sympy/core/expr.py

```diff
@@ -1940,6 +1940,21 @@ def extract_multiplicatively(self, c):
            x/6
 
         """
+        if (type(c) == int and c < 0):
+            return Expr.extract_multiplicatively(self, c*S.ImaginaryUnit*-S.ImaginaryUnit)
```

Comment:
I actually thought just try for -c and return -ans if extract_multiplicatively(-c) returns ans, but i had few failures with integrals i guess and i found it was failing because of integer datatype so when i do c*I*-I it convert c from int to sympy Integer...which fixed the issue.